core/src/com/unciv/ui/worldscreen/bottombar/BattleTable.kt

```diff
@@ -236,11 +237,22 @@ class BattleTable(val worldScreen: WorldScreen): Table() {
         attackerNameWrapper.add(UnitGroup(attacker.unit,25f)).padRight(5f)
         attackerNameWrapper.add(attackerLabel)
         add(attackerNameWrapper)
-
+        var canNuke = true
         val defenderNameWrapper = Table()
         for (tile in targetTile.getTilesInDistance(Battle.NUKE_RADIUS)) {
+
+            //To make sure we dont nuke civilisations we cant declare war with
+            val attackerciv = attacker.getCivInfo()
+            val defenderciv = tile.getCity()?.civInfo
+            if(defenderciv != null && defenderciv.diplomacy[attackerciv.civName]!= null) {
```

Comment:
You can replace the second check with defenderCiv.knows(attackerCiv) for readability

Reply:
> While you made the check for _the tiles_, it is still possible to nuke _the units_ (belonging to a civ you have a peace treaty with) e.g. on the neutral tiles.

oh wow did not consider this, u are totally right!